ocs_ci/utility/kms.py

```diff
@@ -607,18 +607,32 @@ def gather_vault_config(self):
 
     def get_vault_connection_info(self, resource_name=None):
         """
-        Get resource info from ocs-kms-connection-defatils
+        Get resource info from ocs-kms-connection-defatils or csi-kms-connection-details
+        configmap.
 
         Args:
             resource_name (str): name of the resource
 
         """
-        connection_details = ocp.OCP(
+        cm_obj = ocp.OCP(
             kind="ConfigMap",
-            resource_name=constants.VAULT_KMS_CONNECTION_DETAILS_RESOURCE,
             namespace=config.ENV_DATA["cluster_namespace"],
         )
-        return connection_details.get().get("data")[resource_name]
+
+        if cm_obj.is_exist(constants.VAULT_KMS_CONNECTION_DETAILS_RESOURCE):
+            cm_data = cm_obj.get(constants.VAULT_KMS_CONNECTION_DETAILS_RESOURCE)
+            return cm_data.get().get("data")[resource_name]
+
+        elif cm_obj.is_exist(constants.VAULT_KMS_CSI_CONNECTION_DETAILS):
+            cm_data = cm_obj.get(constants.VAULT_KMS_CSI_CONNECTION_DETAILS)
+            for k, v in cm_data.get("data").items():
+                json_out = json.loads(cm_data.get("data")[k])
+                if json_out.get("KMS_SERVICE_NAME") == "vault":
+                    return json_out[resource_name]
+            else:
```

Comment:
seems like wrong indentation for this `else` block

Reply:
This else block is associated only with the for loop. It executes when no KMS_SERVICE_NAME == "vault" is found in the list of connection strings being iterated over in the for loop.